### PR TITLE
fix(lx-select): be able to select an item after scroll

### PR DESCRIPTION
**Pull request by philippeBchn**
When you have a lx-select with a filter, you can't select an item that you see after a scroll in the select. The directive thinks you had clicked out the select (the ng-click of the item is not triggered).

The origin of this bug is the focus on the input. So I blur it when you scroll this the mousewheel in the select (you can't scroll with arrows because the input has the focus and if you click on the scrollbar you focus the right element).

**Step to reproduce**

1. Have a lx-select with enough element to have a scroll and enable the filter and multiple mode.
2. Open the select, you can select a visible item.
3. If you select a visible item, you can scroll and select a new item (= an item which was not visible before you scrolled) because the click on the first item has focused the right element
4. Close the select
5. Open it again, scroll and try to select a new visible element, you can't, the select just closes itself.

With the fix : 
The mousewheel scroll blur the filter input and you can select a new item.

### CODE REVIEW

File: modules/dropdown/js/dropdown_directive.js [hunk 1]
```diff
@@ -800,13 +803,21 @@
 
             focusTimeout = $timeout(function()
             {
-                element.find('input').focus();
+                input.focus();
+                filterHasFocus = true;
             }, 200);
 
+            window.on('mousewheel', function onMousewheel() {
+                if (selectIsFocus) {
```
Comment (clementprevot) [inline modules/dropdown/js/dropdown_directive.js]: `selectIsFocus` ? Tu veux pas dire  `filterHasFocus` ?

**Comment by philippeBchn** [comment]
@clementprevot Exact, j'ai fixé d'autres points dans mon dernier commit.

File: modules/dropdown/js/dropdown_directive.js [hunk 2]
```diff
@@ -807,17 +807,24 @@
...
             }, 200);
 
-            window.on('mousewheel', function onMousewheel() {
-                if (selectIsFocus) {
+            currentWindow.on('mousewheel', function onMousewheel()
+            {
+                if (filterHasFocus)
+                {
                     input.blur();
+                    currentWindow.off('mousewheel');
```
Comment (clementprevot) [inline modules/dropdown/js/dropdown_directive.js]: Attention en faisant ça tu détache **tous** les handler de `mousewheel`.
Du coup il faudra plutôt que tu sorte ta fonction `onMouseWheel` et que tu l'utilise également sur le `off`

File: modules/dropdown/js/dropdown_directive.js [hunk 3]
```diff
@@ -807,17 +807,24 @@
...
 
             scope.$on('$destroy', function()
             {
                 $timeout.cancel(focusTimeout);
                 element.off();
-                currentWindow.off('mousewheel');
+                
+                if (filterHasFocus)
+                {
+                    currentWindow.off('mousewheel');
```
Comment (clementprevot) [inline modules/dropdown/js/dropdown_directive.js]: Même chose

**Pull request closed without merging**